client/state/test/initial-state.js

```diff
@@ -531,6 +531,13 @@ describe( 'initial-state', () => {
 			// we use fake timers from Sinon (aka Lolex) because `lodash.throttle` also uses `Date.now()`
 			// and relies on it returning a mocked value. Jest fake timers don't mock `Date`, Lolex does.
 			clock = useFakeTimers();
+			clearStorageSpy = jest.spyOn( browserStorage, 'clearStorage' ).mockImplementation( () => {
+				return {
+					then( fn ) {
+						fn();
```

Comment:
This mock will call the callback synchronously, which is a behavior different from the real one. A better mock is one of:
```js
.mockImplementation( () => Promise.resolve() );
.mockReturnValue( Promise.resolve() );
.mockResolvedValue(); // my favorite
```

Reply:
yeah, that's ... tricky. We need to run the assertions after all of the promises have resolved, so we need a way to wait until all of the mocked assertions have completed. The sync hack was a hack to work around that. 

Not going to worry about it given we probably have to rework this entirely.